nbsmoke/lint/__init__.py

```diff
@@ -0,0 +1,101 @@
+"""
+Extend pytest to allow flake checking of notebooks.
+
+A notebooks is first converted to a python script, then checked with
+pyflakes.
+
+IPython magics are supposed to be handled in a useful way, being
+converted to python or skipped as appropriate (depending on their
+interaction with python code). See the magics module for more info.
+"""
+
+import sys
+import warnings
+import io
+import os
+import re
+
+import pytest
+import nbformat
+import nbconvert
+
+from . import _pyflakes
+from . import magics
+
+# list of regexes
+flakes_to_ignore = []
+
+class NBLintError(Exception):
+    pass
+
+class LintNb(pytest.Item):
+
+    def repr_failure(self, excinfo):
+        if excinfo.errisinstance(NBLintError):
+            return excinfo.value.args[0]
+        else:
+            return super(LintNb, self).repr_failure(excinfo)
+
+    def runtest(self):
+        with io.open(self.name,encoding='utf8') as nbfile:
+            nb = nbformat.read(nbfile, as_version=4)
+
+            magics_thing = magics.Thing(
+                extra_line_blacklist = _get_list_from_conf('nbsmoke_flakes_line_magics_blacklist', self.parent.parent.config),
+                extra_cell_blacklist = _get_list_from_conf('nbsmoke_flakes_cell_magics_blacklist', self.parent.parent.config))
+            magics_thing.insert_get_ipython(nb)
+
+            ipy, _ = nbconvert.PythonExporter().from_notebook_node(nb)
+
+            debug=self.config.option.nbsmoke_lint_debug
+            filenames = []
+
+            self._write_debug_file(debug, ipy, self.name, "pre", filenames)
```

Comment:
It might make sense to have this be a context manager rather than calling it all over the place.

Reply:
True. It's just twice though, and didn't change in this PR (you're seeing code that's moved from the original single file into its own one, I think).